snap/snapcraft.yaml

```diff
@@ -3,315 +3,135 @@ summary: Live streaming and video recording software
-      - libxau6
+      - libx264-164
+      - libxcb1
       - libxcb-composite0
       - libxcb-randr0
-      - libxcb-render0
-      - libxcb-shape0
       - libxcb-shm0
       - libxcb-xfixes0
       - libxcb-xinerama0
-      - libxcb1
-      - libxcomposite1
-      - libxcursor1
-      - libxdamage1
-      - libxdmcp6
-      - libxext6
-      - libxfixes3
-      - libxi6
-      - libxinerama1
       - libxkbcommon0
-      - libxml2
-      - libxrandr2
-      - libxrender1
-      - libxss1
-      - libxv1
-      - libxvidcore4
-      - libxxf86vm1
-      - libzimg2
-      - libzmq5
-      - libzstd1
-      - libzvbi0
-      - pipewire-bin
-      - ocl-icd-libopencl1
-      - zlib1g
+      - pulseaudio
+      - python3
+      - python3.12
       - qt6-image-formats-plugins
       - qt6-qpa-plugins
       - qt6-wayland
-      - gstreamer1.0-plugins-good
-      - gstreamer1.0-plugins-bad
-      - gstreamer1.0-plugins-ugly
-      - gstreamer1.0-x
-      - libgles2-mesa
-      - libvlc5
-      - vlc-plugin-base
       - stterm
-  cleanup:
-    after: [obs, ndi, launcher]
+      - vlc-plugin-base
+  intel-graphics:
     plugin: nil
-    build-snaps: [gnome-42-2204]
-    override-prime: |
-      set -eux
-      cd /snap/gnome-42-2204/current
-      find . -type f,l -exec rm -f $CRAFT_PRIME/{} \;
+    stage-packages:
+      - vainfo
+      - intel-media-va-driver-non-free
+      - libvpl2
+      - libvpl-dev
+      - libva-drm2
+      - libva-glx2
+      - libva-wayland2
+      - libva-x11-2
+      - libva-dev
+      - libmfx1
+      - libmfx-gen1
+      - libmfx-gen-dev
+      - libigdgmm12
+      - libigdgmm-dev
+
 apps:
   obs-studio:
-    common-id: com.obsproject.Studio
-    extensions: [gnome]
     command: obs-wrapper
+    extensions: [gnome]
     environment:
-      QT_PLUGIN_PATH: $SNAP/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/qt6/plugins:$QT_PLUGIN_PATH
-      LD_LIBRARY_PATH: $LD_LIBRARY_PATH:$SNAP/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/lapack:$SNAP/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/openblas-pthread:$SNAP/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/blas:$SNAP/bin/64bit:$SNAP/obs-plugins/64bit:$SNAP/data/obs-scripting/64bit:$SNAP/cef:$SNAP/lib
-      PIPEWIRE_CONFIG_NAME: $SNAP/usr/share/pipewire/pipewire.conf
-      PIPEWIRE_MODULE_DIR: $SNAP/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/pipewire-0.3
-      SPA_PLUGIN_DIR: $SNAP/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/spa-0.2
+      QT_PLUGIN_PATH: $SNAP/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/qt6/plugins:$QT_PLUGIN_PATH
+      QT_QPA_PLATFORM_PLUGIN_PATH: $SNAP/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/qt6/plugins/platforms
+      LD_LIBRARY_PATH: $SNAP/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/pulseaudio:$SNAP/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/obs-plugins:$SNAP/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/obs-scripting:$SNAP/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/blas:$SNAP/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/lapack:$SNAP/ffmpeg-platform/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR:$LD_LIBRARY_PATH
+      PATH: $SNAP/ffmpeg-platform/usr/bin:$PATH
```

Comment:
When I tested the snap was picking up the iHD driver from the `gpu-2404` content producer snap, which caused libva conflicts and hardware encode to fail on my machine. That snap bundles a command-chain script that sets a `LIBVA_DRIVERS_PATH` variable. I think we can safely prepend to that search path to ensure that the correct version (from our PPA) is loaded first. I attempted this just now on my machine and it worked.

```
LIBVA_DRIVERS_PATH: $SNAP/usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/dri:$LIBVA_DRIVERS_PATH
```

With this change you can also drop the layout, which is a nice bonus. :)

Reply:
thanks! for me both works (even when it's getting the iHD driver from gpu-2404, it then still works with libva 1.22)
but getting rid of the layout is indeed a nice bonus :)